### PR TITLE
Player: Stream Quality panel (redesign of the Statistics popup)

### PR DESCRIPTION
**Pull request by AKnassa**
## What this does

Adds info hash copy and redesigned the network menu slightly

### CODE REVIEW

**Review by Botsy — CHANGES_REQUESTED** [review]
I reviewed and tested the changes. Overall, a clean, well-tested PR. Good call reframing raw numbers into a plain-language verdict. Pure logic extracted into `streamQuality.js` with solid edge-case test coverage; units consistent; conventions followed. 👍

A few non-blocking notes:

1. **Cast path reads "Poor" for healthy streams.** On the Chromecast sender path `buffered` is `null` (vs ms on the HTML path), so `bufferAhead` returns `0` and cast torrents score near-zero on buffer regardless of health. Regression from the old panel. Handle null-buffer or note cast as out of scope.

2. **`keepingUp === null` scored as "not keeping up."** Until `duration` is known (and for live streams), `keepingUp` is `null`, and `keepingUp ? 20 : 0` docks 20 points at the moment the panel opens — healthy streams can transiently read Fair/Poor. Noted as a follow-up; flagging since it hits the primary case.

3. **Minor:** `onCopyInfoHash` deps are `[infoHash]` only — missing `t`/`toast` (stale translation on mid-stream language switch; likely trips `exhaustive-deps`).

4. **Translations:** the inline `defaultValue` strings are fine as a stopgap, but we'd expect the new `PLAYER_QUALITY_*` / `PLAYER_SIGNAL_*` keys to land in a matching PR on the stremio-translations repo before merge.

Thank you for contributing!

**Review by Botsy — DISMISSED** [review]

**Review by kKaskak — CHANGES_REQUESTED** [review]
- Remove tests/streamQuality.spec.js; new tests are prohibited by repo guidance.
- Replace the arbitrary quality score with objective playback signals, or document product-approved thresholds.
- Account for playback speed when calculating buffer runway and required download speed.
 - Show Limited when real player buffer data is unavailable; avoid promises like “will not buffer.”
- Add all strings to stremio-translations, bump the dependency when pr merged, and remove inline English fallbacks.
- Make the new helper TypeScript and remove unrelated/refactored code.
 Remove or split the info-hash copy feature into a separate PR.
- Squash into concise commits without commit bodies and shorten the PR description.
- Verify web, desktop shell, Chromecast, cached/fresh torrents, multiple playback speeds, and non-English UI.

**Comment by kKaskak** [comment]
also for reference we have added a simple contribution guide 
https://github.com/Stremio/stremio-web/blob/development/CONTRIBUTING.md

File: src/routes/Player/useStatistics.js [hunk 1]
```diff
@@ -51,6 +52,29 @@ const useStatistics = (player, streamingServer) => {
             0;
     }, [statistics]);
 
+    const ready = React.useMemo(() => {
```
Comment (kKaskak) [inline src/routes/Player/useStatistics.js]: These are cheap derived values. Could we use plain consts instead of five separate useMemos?

File: src/routes/Player/streamQuality.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2017-2023 Smart code 203358507
+
+export const LOW_BUFFER_RUNWAY_SECONDS = 10;
+
+export type StreamQualityStatus = 'checking' | 'limited' | 'good' | 'fair' | 'poor';
+
+export type StreamQualityReason = 'checking' | 'no-buffer-data' | 'cached' | 'keeping-up' | 'buffer-healthy' | 'draining' | 'buffer-low';
+
+export type StreamQuality = {
```
Comment (kKaskak) [inline src/routes/Player/streamQuality.ts]: status and reason can form invalid combinations. Can we derive one from the other?

File: src/routes/Player/StatisticsMenu/StatisticsMenu.js [hunk 3]
```diff
@@ -4,58 +4,180 @@ const React = require('react');
...
+        setDetailsOpen((open) => !open);
+    }, []);
+    const onToggleKeyDown = React.useCallback((event) => {
+        if (event.key === ' ' || event.key === 'Spacebar') {
+            event.preventDefault();
+            setDetailsOpen((open) => !open);
+        }
+    }, []);
+
+    const bufferValue = cached ?
```
Comment (kKaskak) [inline src/routes/Player/StatisticsMenu/StatisticsMenu.js]: These nested ternaries are hard to scan. A small formatter or regular if block would be clearer.

File: src/routes/Player/streamQuality.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    if (!ready) {
+        return { status: 'checking', reason: 'checking' };
+    }
+    if (cached) {
+        return { status: 'good', reason: 'cached' };
+    }
+    if (bufferRunway === null) {
+        return { status: 'limited', reason: 'no-buffer-data' };
+    }
+    if (keepingUp === true) {
```
Comment (kKaskak) [inline src/routes/Player/streamQuality.ts]: This can report Good with an empty buffer. The low-buffer check should happen first.

File: src/routes/Player/StatisticsMenu/StatisticsMenu.js [hunk 5]
```diff
@@ -4,58 +4,180 @@ const React = require('react');
...
-                <div className={styles['stat']}>
-                    <div className={styles['label']}>
-                        {t('PLAYER_PEERS')}
-                    </div>
-                    <div className={styles['value']}>
-                        { peers }
-                    </div>
+            <div className={styles['header']}>
+                <div className={styles['title']}>
+                    {t('PLAYER_STREAM_QUALITY')}
```
Comment (kKaskak) [inline src/routes/Player/StatisticsMenu/StatisticsMenu.js]: These keys aren’t in the pinned translations yet, so users will see the raw key names. please make coresponding pr if you didnt pls link it here as a comment so we can merge it.

**Review by kKaskak — CHANGES_REQUESTED** [review]
also check for repetetive if statements there is a lot of logic that could be simplifed a lot, and a lot of checks we dont really need impl could be much simpler

File: src/routes/Player/useStatistics.js (hunk 1 above)
Comment (AKnassa) [reply src/routes/Player/useStatistics.js]: done, plain consts now

File: src/routes/Player/streamQuality.ts (hunk 2 above)
Comment (AKnassa) [reply src/routes/Player/streamQuality.ts]: done, single string now, the component derives tone/label/verdict from one map

File: src/routes/Player/StatisticsMenu/StatisticsMenu.js (hunk 3 above)
Comment (AKnassa) [reply src/routes/Player/StatisticsMenu/StatisticsMenu.js]: done, small formatters

File: src/routes/Player/streamQuality.ts (hunk 4 above)
Comment (AKnassa) [reply src/routes/Player/streamQuality.ts]: right, fixed. low-buffer check runs first now. fresh seeks read Poor until runway passes 10s

File: src/routes/Player/StatisticsMenu/StatisticsMenu.js (hunk 5 above)
Comment (AKnassa) [reply src/routes/Player/StatisticsMenu/StatisticsMenu.js]: linked: https://github.com/Stremio/stremio-translations/pull/1080, rebased on latest master, conflict resolved, mergeable now

**Review by Botsy — COMMENTED** [review]
@kKaskak since your changes override completely the original idea behind this PR, can you please move your changes to a separate PR that contains only them? If you think this PR's original solution is not to be merged, please close this PR.

**Review by Botsy — APPROVED** [review]
@kKaskak tested and looks good 👍

**Review by kKaskak — APPROVED** [review]
